react-app/src/components/ResultSidebar.js

```diff
@@ -1,18 +1,35 @@
-import React from 'react';
-import { Button, Accordion } from 'react-bootstrap';
+import React, { useState } from 'react';
+import { Alert, Button, Accordion } from 'react-bootstrap';
 import { useHistory } from 'react-router-dom';
 import ResultHeader from './ResultHeader';
 import Result from './Result';
 
 function ResultSidebar(props) {
   const history = useHistory();
+  const [showSavedAlert, setShowSavedAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [showNoFilesAlert, setShowNoFilesAlert] = useState(false);
+
   function onExitResultPage() {
     // reset data values and return to home page
     history.push('/extract');
   }
 
   function onSave() {
-    // Save the results permanently somehow
+    if (props.extractedData.length > 0) {
+      window.api.getOutputPath().then((savePath) => {
+        if (!savePath.canceled) {
+          const result = window.api.saveOutput(savePath.filePaths[0], props.extractedData);
+          if (result) {
+            setShowSavedAlert(true);
+          } else {
+            setShowErrorAlert(true);
+          }
```

Comment:
I don't think this block is doing what it appears. The `saveOutput` function is an `async` function that eventually resolves to a boolean value from the `save-output` IPC message. You either need to use `await` or chain another `.then` (do not nest `.then`s inside each other, this is an anti pattern. Read up on promise chaining here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#chaining)

Right now, `result` is actually an unresolved promise that _might_ be `true` or `false`, but it will always pass the first `if` check because the Promise is never awaited or `.then`ed

Reply:
Thanks for the resource on promise chaining! That clarified a lot of things